pkg/destroy/azure/azure.go

```diff
@@ -359,12 +406,28 @@ func wasNotFound(resp *http.Response) bool {
 	return resp != nil && resp.StatusCode == http.StatusNotFound
 }
 
+func isAuthError(err interface{}) bool {
+	switch e := err.(type) {
+	case autorest.DetailedError:
+		switch statusCode := e.StatusCode.(type) {
+		case int:
+			if statusCode >= 400 && statusCode <= 403 {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
something on the lines of

```suggestion
func isAuthError(err interface{}) bool {
	if err == nil {
		return false
	}

	var dErr autorest.DetailedError
	if errors.As(err, &dErr)  {
		switch statusCode := dErr.StatusCode.(type) {
		case int:
			if statusCode >= 400 && statusCode <= 403 {
				return true
			}
		}
	}
	return false
}
```